### PR TITLE
Adding app

### PR DESCRIPTION
**Pull request by ghost**
Definitely take a look at the **possible use cases**; couldn't think of anything good.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+**Possible Use Cases**
+  * Adding gender classification to a face detection system
+
+
+## Input Scheme
+The input should contain a base64 encoded image of a face. 
+``` json
+{
+  "face": "aGVsbG8gZnJvbSB0aGUgb3RoZXJzaWRl"
```
Comment (bvallelunga) [inline README.md]: change `face` to `image` . It is important to create a standard interface.


change `aGVsbG8gZnJvbSB0aGUgb3RoZXJzaWRl` to `BASE_64_ENCODED_IMAGE`

File: README.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+}
+```
+
+## Output Scheme
+The output will map each input text to a score. Positive receive positive scores while negative texts receive negative 
+scores. 
+ 
+``` json
+{
+  "scores":
```
Comment (bvallelunga) [inline README.md]: change output scheme to:

``` js
{
   "woman": 0.91,
   "man": 0.09
}

File: package.json [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  "website": "https://github.com/oarriaga/face_classification",
+  "repository": "https://github.com/DopplerFoundation/example-app-gender",
+  "input_scheme": {
+    "face": "string"
+  },
+  "output_scheme": {
+    "scores": "object"
+  },
+  "license": "Apache License",
+  "dependencies": [
```
Comment (bvallelunga) [inline package.json]: No need to add `dependencies` in package.json

File: package.json [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  "keywords": "gender classification computer vision face",
+  "description": "Quickly determine the gender of a face. Trained on hundreds of thousands of imdb images.",
+  "main": "app/main.py",
+  "website": "https://github.com/oarriaga/face_classification",
+  "repository": "https://github.com/DopplerFoundation/example-app-gender",
+  "input_scheme": {
+    "face": "string"
+  },
+  "output_scheme": {
+    "scores": "object"
```
Comment (bvallelunga) [inline package.json]: change to 

``` js
"output_scheme": {
  "man": "number",
  "woman": "number"
}
```

**Review by bvallelunga — CHANGES_REQUESTED** [review]

**Comment by bvallelunga** [comment]
**additional use cases**

- targeted messaging to your users based on their gender/interests
   - especially useful if you do not ask what a users gender is during registration
- categorizing profiles from social networks by gender

File: app/main.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+            face = base64_to_img(input['face'])
+        except (binascii.Error, AttributeError):
+            raise ValueError("'face' must be a base64 encoded string.")
+        min_dim = self.model.input_shape[1]
+        if any(dim < min_dim for dim in face.size):
+            raise ValueError("'face' can not have a height or width less than {} pixels.".format(min_dim))
+
+        face = preprocess(face, self.model.input_shape[1:3])
+        scores = self.model.predict(face).tolist()[0]
+        return {'scores': {label: round(score, SCORE_PRECISION) for label, score in zip(LABELS, scores)}}
```
Comment (ghost) [inline app/main.py]: make sure class predictions sum to 1 via 1 - class_score_1 = class_score_2

**Review by bvallelunga — APPROVED** [review]

**Pull request merged**